FontSizePicker: Fix FontSizePicker Storybook control type and improve documentation (#68936)
WordPress/gutenberg: packages/components/src/font-size-picker/README.md

Available units for custom font size selection.

> **Important Note**: For the units property to work, font sizes must be specified as strings with units (e.g., `'12px'` instead of `12`). When font sizes are provided as numbers, the component operates in "unitless mode" where the units property has no effect.

Reviewer: ```suggestion
**Note**: For the `units` property to work, the current font size value must be specified as strings with units (e.g., `'12px'` instead of `12`). When the font size is provided as a number, the component operates in "unitless mode" where the `units` property has no effect.
```

My feeling is that this should be emphasized in the `value` prop section, what do you think?

Author: Yes @t-hamano you're right, It makes more sense to emphasize the `value` prop selection.

> **Note**: For the `units` property to work, the current font size value must be specified as strings with units (e.g., `'12px'` instead of `12`). When the font size is provided as a number, the component operates in "unitless mode" where the `units` property has no effect.

Reviewer: What do you think about moving this to [the `value` prop section](https://github.com/WordPress/gutenberg/blob/7a4b5aa5d8a63d97b084e1339e6ce544dfc944bb/packages/components/src/font-size-picker/README.md#value-number--string)? Because unitless mode is based on the `value` prop value, not the `units` prop value.

Reviewer 2: Let's also make sure to add these same text to the JSDocs in the `units` prop type
